README.md

```diff
@@ -22,26 +24,36 @@ cd pages-starter-react-locations
 npm install
 ```
 
-### Useful commands
+### Recommended Development Flow
 
-`yext init` - Authenticates the Yext CLI with your Yext account
+While _developing locally_, run the following command:
+
+```
+npm run dev
+```
+
+  This command will start a Vite-powered dev server that will enable hot-reloading. Additionally, the command will generate a `localData` directory that contains a subset of your Knowledge Graph data. This command is automatically in "dynamic" mode, which means it will pull data updates automatically from your Knowledge graph, so real-time data changes in your Yext account will be reflected in your local dev site.
 
-`npm run dev` - runs your code against a local dev server using Vite
+  NOTE: Whenever you make changes to your stream definitions, you must re-run `npm run dev` for the system to updates the `features.json` and the required entities to power your site. 
 
-- All stream documents come from the `localData` folder
-- You can visit either of these urls out of the box
-  - http://localhost:3000/index/123
-  - http://localhost:3000/static
+_Before committing_ your code, we recommend running the following command:
+```
+npm run build:serve
+``` 
+
+  This command will generate a production build of your site, so you can ensure there are no build errors or unexpected behavior. This build step replicates the production build environment used in the Yext system, and serves your data at `localhost:8000`.
+
+  In practice, development builds (via `npm run dev`) and production builds compile and bundle assets differently. For local development, ES Modules are loaded directly by the browser, allowing fast iteration during local development and also allows for hot module replacement (HMR). Other things like CSS are also loaded directly by the browser, including linking to sourcemaps. During a production build all of the different files are compiled (via ESBuild for jsx/tsx) and minified, creating assets as small as possible so that the final html files load quickly when served to a user.
```

Comment:
You might also want to mention "tree-shaken" in the compiled/minified sentence

Reply:
@mkilpatrick can you give more detail about what you'd like me to mention? I just want to make sure I'm accurate in what I"m saying.

Reply:
tree-shaking happens during bundling, which is the process of removing any dependencies that aren't actually used. I was just saying to mention it here. Most devs should know what it means.